src/iptvsimple/utilities/WebUtils.cpp

```diff
@@ -45,13 +44,13 @@ const std::string WebUtils::UrlEncode(const std::string& value)
 std::string WebUtils::ReadFileContentsStartOnly(const std::string& url, int* httpCode)
 {
   std::string strContent;
-  void* fileHandle = XBMC->OpenFile(url.c_str(), 0x08); //READ_NO_CACHE
-  if (fileHandle)
+  kodi::vfs::CFile file;
+
+  if (file.OpenFile(url, READ_NO_CACHE))
   {
     char buffer[1024];
-    if (int bytesRead = XBMC->ReadFile(fileHandle, buffer, 1024))
+    if (int bytesRead = file.Read(buffer, 1024))
```

Comment:
How come you don't close the file? Or does it auto close when destroyed?

Reply:
It makes by destructor a auto close of file. Only sometimes I add .Close() it if the CFile use is on begin and bigger work behind his work or if CFile used for another file.